symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageUtil.java

```diff
@@ -0,0 +1,76 @@
+package com.symphony.bdk.core.service.message.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apiguardian.api.API;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class for pre-processing an outgoing message
+ */
+@API(status = API.Status.EXPERIMENTAL)
+public class MessageUtil {
+  private static Map<String, String> tokens = initializeMap();
+  private static final Pattern pattern = initializePattern(tokens);
+
+  private MessageUtil() {
+  }
+
+  /**
+   * This method takes care of all special characters placed within the messageML that must be HTML-escaped
+   * to have a valid MessageML format.
+   *
+   * @param rawText text to be parsed
+   * @return text in a valid messageML format
+   */
+  public static String escapeSpecialChars(String rawText) {
+    Matcher matcher = pattern.matcher(rawText);
+    StringBuffer parsedText = new StringBuffer();
+    while (matcher.find()) {
+      matcher.appendReplacement(parsedText, getReplacementOf(matcher.group(1)));
+    }
+    matcher.appendTail(parsedText);
+    return parsedText.toString();
+  }
+
+  private static String getReplacementOf(String group) {
+    String replacement = tokens.get(group);
+    if (replacement == null) {
+      replacement = tokens.get("\\" + group);
+    }
+    return replacement;
+  }
+
+  private static Map<String, String> initializeMap() {
```

Comment:
- Is there any third party library that could do that for us ?
- Why do we have double slashes ? I don't really understand. Do we expect our users to add double slashes before the special characters below ? As a developer, I would expect being able to put my raw string without "escaping" slashes.

Reply:
maybe StringEscapeUtils ? from apache commons text

Reply:
We have all those double slashes to build the regex for the pattern otherwise some of them will be considered as part of the regex and not matched (like: []{}.*). But at the end the developer doesn't need to add the slashes in the raw text.

I tried to use the StringEscapeUtils to escape characters but is not working for all of them. Also I was not able to find a full list of which chars we should really escape that's why I basically kept same behaviour has it was done for the legacy BDK. Only information I found is here: https://developers.symphony.com/symphony-developer/docs/messagemlv2#root-tag, but this is not the full list.